### DISCUSSION TITLE
Add support for image embeds in chat overlay

### DISCUSSION THREAD

**Discussion by darkralts** [ppy/osu#13168] [Ideas]
**Describe the new feature:** 
when you send an osu screenshot (eg https://osu.ppy.sh/ss/6000000) to a friend or maybe even in a chat, have an option for embedding it 
**Proposal designs of the feature:**

**Comment by LittleEndu** [comment]
Disagree with embedding it, that just isn't what the chat is for. If anything it could load a preview when hovering over the link.

**Comment by enoslayd** [comment]
What if when screen shot link was clicked there will be an inclient viewer?

**Reply by sr229** [reply]
I honestly disagree making this an option since osu is not a social application, the social features are there to allow you to communicate via text chat, but if you really just want to share things there, use Discord or something else for that.

**Reply by Konpaku-LVX** [reply]
> I honestly disagree making this an option since osu is not a social application, the social features are there to allow you to communicate via text chat, but if you really just want to share things there, use Discord or something else for that.

it kind of is though

**Reply by frenzibyte** [reply]
I see no problem with having an in-client image viewer, along with markdown rendering. It would be bad UX if you can't click on the image to fill the screen and look at it in-depth.

**Comment by darkralts** [comment]
ah that would be cool,
i think i like that idea better actually :D

**Comment by Game4all** [comment]
I think that before worrying about how screenshots links should be rendered in the client, we should probably assessing first whether we want to  port to lazer the ability to upload screenshots present on stable.

**Reply by Konpaku-LVX** [reply]
it will be added, peppy said he wants to make lazer so good there is 0 reason to use the old client, and that would include that feature, if not it wouldnt match the 0 reason quota.

**Reply by Game4all** [reply]
I have opened an issue to track this addition (#10718 for reference).

**Comment by enoslayd** [comment]
Thats given

**Comment by LittleEndu** [comment]
In-client image viewer with preview on hover that supports all images from known safe hosting sites would be the way to go. Again, I disagree with embedding it (unless it was made an option, but that probably won't happen) and would like it if the thread title was changed to better reflect what people have suggested here.

**Reply by gagahpangeran** [reply]
How about markdown support for image in chat?

**Comment by darkralts** [comment]
I kind of meant for osu.ppy.sh screenshots, I think embeds would be pretty
clean, and if it's just osu screenshots it would be safe, it's peppys
server after all! :) hovering sounds like the way to go for everything else
though

**Comment by developomp** [comment]
Since the image might be hosted in a malicious website (for IP gathering or smth), this feature should be limit only to `osu.ppy.sh` domain.

**Comment by gagahpangeran** [comment]
I think the general direction of this idea is support for markdown in chat (https://github.com/ppy/osu/issues/9145).

With this feature, you can embed image from any image hosting with markdown syntax.
